src/umpire/tpl/CMakeLists.txt

```diff
@@ -114,7 +114,6 @@ if (NOT TARGET camp)
     blt_register_library(
       NAME camp
       DEPENDS_ON cuda_runtime
-      LIBRARIES camp
```

Comment:
@adayton1 - why is this removed?

Reply:
Doesn't this try to add -lcamp to the link line?

Reply:
@white238 @gzagaris camp is a header only library, so should `LIBRARIES camp` be left out of the call to `blt_register_library`?

Reply:
So it does a bunch of possible things:

1) If it is a BLT object library (which is done by blt_add_library(...OBJECT TRUE...)) it circumvents the normal CMake object libraries that have a lot of side effects that are very unhelpful.

2) If it is a BLT registered library, it adds all the known information from the previous blt_register_library

(the one its probably doing) 3) The normal case calls target_link_library() (which is a normal CMake function), which if its a CMake target, inherits information following the CMake normal rules, or if it is not a CMake target, it blindly adds -lcamp to the link line in hopes that it will find it in the system or an already added -L directory.

If you need more clarification I can look into it further.

Reply:
@white238 - camp provides a proper CMake target regardless of whether it's "external" or not. The only extra requirement is the `CAMP_HAVE_CUDA` define to make sure it enables stuff that's guarded in the headers.

What's the best way to do this?